File: programmers/js/level01/12935.js
// 제일 작은 수 제거하기
function solution(arr) {
  const min = Math.min(...arr);
  const result = arr.filter(num => num !== min);
  
  return result.length ? result : [-1];
}

// 다른 풀이
// function solution(arr) {
//   arr.splice(arr.indexOf(Math.min(...arr)), 1);

//   if(arr.length < 1) return [-1];

//   return arr;
// }

File: programmers/java/level01/12935.java
// 제일 작은 수 제거하기

import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr) {
        if (arr.length == 1) return new int[]{ -1 };
        
        int min = Arrays.stream(arr).min().getAsInt();
        
        return Arrays.stream(arr)
            .filter(num -> num != min)
            .toArray();
    }
}